pkg/providers/vsphere/vsphere.go

```diff
@@ -1072,6 +1073,18 @@ func BuildTemplateMap(clusterSpec *cluster.Spec, datacenterSpec v1alpha1.VSphere
 		values["cgroupDriverSystemd"] = true
 	}
 
+	if clusterSpec.Spec.ECRMirror != nil {
+		values["ecrMirrorEndpoint"] = clusterSpec.Spec.ECRMirror.Endpoint
+		if clusterSpec.Spec.ECRMirror.CACert != "" {
+			cert, err := ioutil.ReadFile(clusterSpec.Spec.ECRMirror.CACert)
```

Comment:
Maybe I'm mistaken, but I believe we were trying to avoid having paths to local disk in the cluster spec. Maybe we can just have the content and like we do for the public ssh keys?
@vivek-koppuru you might have opinions on this?

Reply:
We probably don't want the entire cert dumped in the clusterconfig. It looks ugly and there can be indentation issues. For example, this is the cert file for https://anywhere.eks.amazonaws.com/
```
-----BEGIN CERTIFICATE-----
MIIF6TCCBNGgAwIBAgIQAXMVwQseVSA7gsjh+2rUkTANBgkqhkiG9w0BAQsFADBG
MQswCQYDVQQGEwJVUzEPMA0GA1UEChMGQW1hem9uMRUwEwYDVQQLEwxTZXJ2ZXIg
Q0EgMUIxDzANBgNVBAMTBkFtYXpvbjAeFw0yMTA5MDcwMDAwMDBaFw0yMjEwMDYy
MzU5NTlaMCUxIzAhBgNVBAMTGmFueXdoZXJlLmVrcy5hbWF6b25hd3MuY29tMIIB
IjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAzDF5cKPXcFzNZBK4DnMBgrRe
hxoMjT91PTD+6Yn0pyPCaVWRbiNw+TNmuYHJJMZ7RWbe9BX+BI06dhI77HmzX0YX
nnNsLqbFb5Z9InESrFaXdi0PMyXx6XJYK3U4QngUZB4j+t2KxibnSNWxKmEF+OD+
CTSGUkhPqghL9FwTqnLi8hG9dRMIrX5pA+c4c37NrU7Xsb6Uj3+54l3qYzZJd5uX
ifdqktI80RIp3UqtSKoFyxLSY3nUZeWqQCzZ8FCUSEI7v0POBPLgOmwY4JzYvc4E
+h7MbqAHM4f9ynPid7RDBcNfaSLWVeL6VJqJLzfzgor5Xla8KCv2tck1yrYiiwID
AQABo4IC8jCCAu4wHwYDVR0jBBgwFoAUWaRmBlKge5WSPKOUByeWdFv5PdAwHQYD
VR0OBBYEFOfr+xA6J1/QCOOG3m6btNOeTCKxMCUGA1UdEQQeMByCGmFueXdoZXJl
LmVrcy5hbWF6b25hd3MuY29tMA4GA1UdDwEB/wQEAwIFoDAdBgNVHSUEFjAUBggr
BgEFBQcDAQYIKwYBBQUHAwIwOwYDVR0fBDQwMjAwoC6gLIYqaHR0cDovL2NybC5z
Y2ExYi5hbWF6b250cnVzdC5jb20vc2NhMWIuY3JsMBMGA1UdIAQMMAowCAYGZ4EM
AQIBMHUGCCsGAQUFBwEBBGkwZzAtBggrBgEFBQcwAYYhaHR0cDovL29jc3Auc2Nh
MWIuYW1hem9udHJ1c3QuY29tMDYGCCsGAQUFBzAChipodHRwOi8vY3J0LnNjYTFi
LmFtYXpvbnRydXN0LmNvbS9zY2ExYi5jcnQwDAYDVR0TAQH/BAIwADCCAX0GCisG
AQQB1nkCBAIEggFtBIIBaQFnAHYAKXm+8J45OSHwVnOfY6V35b5XfZxgCvj5TV0m
XCVdx4QAAAF7wc5GyAAABAMARzBFAiEAmsvu7lEM+dMOh355tJ89y2iOhun4oMkK
OSaZyAZgMdUCIDaGPwP8SfYSatAclTyrU0MS/4OWKTpzXW9ub1FJCzGuAHUAQcjK
sd8iRkoQxqE6CUKHXk4xixsD6+tLx2jwkGKWBvYAAAF7wc5HIAAABAMARjBEAiBU
Crrn7iWcUkyUPPw83KIJxJ6XclzgYJW97AK3d20mzwIgbQ/Ykg0IxAeCuSw5DOFO
X2/JrSVqb/pGzYi/bLSxFNsAdgDfpV6raIJPH2yt7rhfTj5a6s2iEqRqXo47EsAg
RFwqcwAAAXvBzkcLAAAEAwBHMEUCICPvgyay2qyYIniszuK4YooJ9YzWjc5ejde1
aQmDvH6lAiEAnLwbDDidvdslWmTOQIWOnIaPFfGC29CGQi7b7RA1iTowDQYJKoZI
hvcNAQELBQADggEBAJzQxIJXpSY/4Xs5IasifbSW9RnN7QGV0M/FYgP2Pi5u6mEg
jXW1bwnlZ+0j45Ee06gwFSTUchW7Om5zfS/5brk23yDnLK7ENpgtUgykGe6518oF
ODagzBvwPiQn56SN+USqQueuoehmZvTi1rbPDxpctQl02zVaTODjn9sgF8yUS2F1
qQ2VR5u+1okDuA6ofdrWMUfJpDcREO0FsFOn9B9fltm5m2/Y1oO5vG2QhaKQ3NqT
j8kWVdmjehuwhC05x0uPN+m/TuGlPvZ9ji87rs58fmPt0rRBcPBAKYSn53KNpnUO
czPpLexr0/DE8JejN8Y9lSK5zsAqWk6gGRS5AFg=
-----END CERTIFICATE-----
```

Reply:
I agree with what Guillermo said above as path's won't make from the Kubernetes perspective once the spec is applied. Content should be okay right?

Reply:
Sorry looks like the comments loaded after I posted the comments. Can we pass it in through an environment variable, and then store it on the Kubernetes cluster through a certificate or a secret that can be referenced from the Cluster spec? This is similar to what we had in mind for other similar fields including this issue: https://github.com/aws/eks-anywhere/issues/166

Reply:
How about having a base64 encoded cert as the string in the cluster object?